Tests/HttpExceptionFactoryTest.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Exceptions\Tests;
+
+use Exceptions\Data\ValidationException;
+use Exceptions\Http\Client;
+use Exceptions\Http\Server;
+use Exceptions\Helpers\HttpExceptionFactory;
+use Exceptions\Tests\stub\CustomHttpException;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+
+class HttpExceptionFactoryTest extends TestCase
+{
+    /**
+     * @dataProvider dataProvider
+     * @param int $errorCode
+     * @param string $expectedClass
+     */
+    public function testBuild(int $errorCode, string $expectedClass)
+    {
+        $this->assertInstanceOf($expectedClass, HttpExceptionFactory::build($errorCode));
+    }
+
+    public function testBuildFailed()
+    {
+        $this->expectException(ValidationException::class);
+        HttpExceptionFactory::build(101);
+    }
+
+    /**
+     * @dataProvider dataProvider
+     * @param int $errorCode
+     * @param string $expectedClass
+     */
+    public function testBuildWithContext(int $errorCode, string $expectedClass)
+    {
+        $ctx = new stdClass();
+        $exception = HttpExceptionFactory::buildWithContext($errorCode, $ctx);
+        $this->assertInstanceOf($expectedClass, $exception);
+        $this->assertSame($ctx, $exception->getContext());
+    }
+
+    public function testBuildWithContextFailed()
+    {
+        $this->expectException(ValidationException::class);
+        HttpExceptionFactory::buildWithContext(101, []);
+    }
+
+    public function testExtendingHelper()
+    {
+        $helper = new class extends HttpExceptionFactory {
```

Comment:
Wow, i didn't know this was possible in PHP. I completely missed that in feature introduction...

Reply:
PHP 7.0 if I'm right